index.js

```diff
@@ -42,22 +56,38 @@ function walkSync (baseDir, _options, _relativePath) {
   var entries = fs.readdirSync(baseDir + '/' + relativePath).sort();
   for (var i = 0; i < entries.length; i++) {
     var entryRelativePath = relativePath + entries[i];
-    var stats = getStat(baseDir + '/' + entryRelativePath);
+    var fullPath = baseDir + '/' + entryRelativePath;
+    var stats = getStat(fullPath);
 
     if (stats && stats.isDirectory()) {
       if (options.directories !== false && (!m || m.match(entryRelativePath))) {
-        results.push(entryRelativePath + '/');
+        results.push(new Entry(relativePath, stats.mode, stats.size, stats.mtime.getTime()));
```

Comment:
Hm wait, I think `stats` can be undefined, so `stats.mode` might throw an error.